api/matlab/arrus/mexcuda/rawReorg.cu

```diff
@@ -156,8 +158,17 @@ void mexFunction(int nlhs, mxArray * plhs[],
         out = mxGPUCreateGPUArray(nDimOut, dimOut, mxSINGLE_CLASS, mxCOMPLEX, MX_GPU_INITIALIZE_VALUES);
         float2 * dev_out = static_cast<float2 *>(mxGPUGetData(out));
         
-        dim3 blocksPerGrid = {1, nSampIn / 2 / threadsPerBlock.y, 1};
-        rawReorgCplx<<<blocksPerGrid, threadsPerBlock>>>( dev_out, dev_in, dev_reorgMap, nSampOut);
+        /* Kernel in loop - due to blocksPerGrid limit */
+        int nBlock = nSampIn / 2 / threadsPerBlock.y;
+        int nBlockPerPart = (nBlock <= 65535) ? nBlock : 65535;
+        int nPart = (nBlock+nBlockPerPart-1)/nBlockPerPart;
+        for (int iPart=0; iPart<nPart; iPart++) {
+            
+            unsigned int nBlockInThisPart = (iPart<(nPart-1)) ? nBlockPerPart : (nBlock-iPart*nBlockPerPart);
+            
+            dim3 blocksPerGrid = {1, nBlockInThisPart, 1};
```

Comment:
@PiotrKarwat why we are limiting to a 1D grid here?

Reply:
@pjarosik the data is 32 x X x 1, threadBlock is 32 x 32 x 1, so the grid is 1 x (X/32) x 1. We could use 2D grid and convert its indices into 1D in the kernel, but why add anything extra to the kernel? 
Regarding your next comment ("Also, the max size of the grid can be read from the CUDA device props"):
- Max size of the grid is the same for all compute capabilities, so there is no need to check it... for now. Future compute capabilities may, of course, introduce some changes.
- Your comment made me realize that instead of {1, nBlock, 1} I can use {nBlock, 1, 1} grid - the max x-size of a grid is much higher than y- or z-size (2^31-1 vs 2^16-1). This solves the problem without calling the kernel in the for loop. Changes are underway.

Reply:
@pjarosik new solution is implemented. Take a look and let me know which version you prefer.